coursefunctions.h

```diff
@@ -449,43 +449,48 @@ void TestVertexShader(Buffer2D<PIXEL> & target)
 
         PIXEL colors[3] = {0xffff0000, 0xff00ff00, 0xff0000ff};
         // Your code for 'colorAttributes' goes here
+        colorAttributes[0].addAttributes(1,0,0);
+        colorAttributes[1].addAttributes(0,1,0);
+        colorAttributes[2].addAttributes(0,0,1);    
 
         FragmentShader myColorFragShader;
+        myColorFragShader = ColorFragShader;
+
+        Attributes colorUniforms; 
 
-        Attributes colorUniforms;
-        // Your code for the uniform goes here, if any (don't pass NULL here)
-        
         VertexShader myColorVertexShader;
-        // Your code for the vertex shader goes here 
+        myColorVertexShader = VertexShader(&TransformationVertShader);
+
 
         /******************************************************************
-		 * TRANSLATE (move +100 in the X direction, +50 in the Y direction)
+         * TRANSLATE (move +100 in the X direction, +50 in the Y direction)
          *****************************************************************/
         // Your translating code that integrates with 'colorUniforms', used by 'myColorVertexShader' goes here
-
-		DrawPrimitive(TRIANGLE, target, colorTriangle, colorAttributes, &colorUniforms, &myColorFragShader, &myColorVertexShader);
+        colorUniforms.addAttributes(100, 50);
+	DrawPrimitive(TRIANGLE, target, colorTriangle, colorAttributes, &colorUniforms, &myColorFragShader, &myColorVertexShader);
 
         /***********************************
          * SCALE (scale by a factor of 0.5)
          ***********************************/
         // Your scaling code that integrates with 'colorUniforms', used by 'myColorVertexShader' goes here
-
+        colorUniforms.addAttributes(0,0,0,0,0.5,0.5,0);
```

Comment:
I like how fast this is. I have like 10 lines of code for this one. Pretty smart. It makes it a bit less readable for reviewers like me but that doesn't really matter I guess!